bee-hive/Dockerfile

```diff
@@ -0,0 +1,54 @@
+ARG MICRO_IMAGE_DIR=/ubi-micro-img
+
+# BASE image using UBI 9 micro where the
+# application and requirements will be installed
+FROM registry.access.redhat.com/ubi9-micro:9.4-15 AS base
+
+# BUILD image using UBI 9 where the dependencies that
+# require installing with a package manager will be installed
+FROM registry.access.redhat.com/ubi9:9.4-1214.1726694543 AS build
+ARG MICRO_IMAGE_DIR
+
+# Copy the BASE image into the BUILD image
+RUN mkdir ${MICRO_IMAGE_DIR}
+COPY --from=base / ${MICRO_IMAGE_DIR}
+
+# Install Python inside the BASE image
+RUN dnf install --installroot ${MICRO_IMAGE_DIR} --nodocs -y \
+    python3.11-3.11.9 \
+    python3.11-devel-3.11.9 \
+    libstdc++ &&\
+    dnf upgrade --installroot ${MICRO_IMAGE_DIR} --nodocs -y && \
+    dnf clean all --installroot ${MICRO_IMAGE_DIR}
+
+# APP image from `scratch` which will be the final image
+# and remaining application requirements will be installed
+FROM scratch AS app
+ARG MICRO_IMAGE_DIR
+COPY --from=build ${MICRO_IMAGE_DIR}/ .
+
+# create symlinks for python
+RUN ln -s /usr/bin/python3.11 /usr/bin/python
+
+# Create project dir
+WORKDIR /usr/src/app
+
+# set environment variables
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+
+# Install pip
+RUN python3.11 -m ensurepip --upgrade
+# Install dependencies and update then uninstall pip (not needed in final image)
+RUN python3.11 -m pip install openai pyyaml python-dotenv --no-cache-dir --upgrade && \
```

Comment:
In future can we use poetry to do the install, then it will be consistent across local install & in container, and be up to date with required dependencies - obviously python (3.11 or 3.12) will need to be installed, and poetry (pipx)
Not a concern for early versions when we're iterating to get something working though.